src/sdk/reencrypt.test.ts

```diff
@@ -1,28 +1,28 @@
-import {
-  FheUint160,
-  CompactFheUint160List,
-  TfheCompactPublicKey,
-  TfheClientKey,
-} from 'node-tfhe';
 import { reencryptRequest } from './reencrypt';
-import { createTfheKeypair } from '../tfhe';
-import { createEncryptedInput } from './encrypt';
-import { getPublicKeyCallParams, getPublicKeyFromNetwork } from './network';
-import { fromHexString } from '../utils';
+import fetchMock from 'fetch-mock';
+
+global.fetch = require('fetch-mock-jest').sandbox();
 
 describe('reencrypt', () => {
   it('get reencryption for handle', async () => {
-    const reencrypt = reencryptRequest('mock');
-    const result = await reencrypt(
-      BigInt(3333),
-      '0xccc',
-      '0xcccc',
-      '0xccc',
-      '0x8ba1f109551bd432803012645ac136ddd64dba72',
-      '0xa5e1defb98EFe38EBb2D958CEe052410247F4c80',
-    );
-
-    expect(result.toString()).toBe('10');
+    // const reencrypt = reencryptRequest('http://mock');
+    // fetchMock.mock('http://mock/reencrypt', {
+    //   status: 200,
+    //   body: {
+    //     response: {
+    //       mock: 's',
+    //     },
+    //   },
+    // });
+    // const result = await reencrypt(
+    //   BigInt(3333),
+    //   '0xccc',
+    //   '0xcccc',
+    //   '0xccc',
+    //   '0x8ba1f109551bd432803012645ac136ddd64dba72',
+    //   '0xa5e1defb98EFe38EBb2D958CEe052410247F4c80',
+    // );
+    // expect(result.toString()).toBe('10');
```

Comment:
I don't fully understand what's in the mock, but it should work if the gateway send back something like:
```
{
"response":
[
  {
    version: 1,
    servers_needed: 1,
    verification_key: '210000000000000002be83edb0000dab98dc294dcd3bd030254b009f44271c1e4c600b3d2642028100',
    digest: '65d06bfdcc1922d55f755ea17f9390b043ae6905cb40634b8236f4383877a12b204c066f562e6753b7e8bbdb8b0034fcfd76dcf4642a4ec8aec7378dd4665ac3',
    fhe_type: 'euint8',
    signcrypted_ciphertext: 'ed000000000000002d137db51b420bb52dd50c3e492b8e2fd45471cf48c30cfc8446169ef838c07dd670edece22f47e106bb6a935f7a8d5b203b5bb27f832b7c1d6bac3336b1258041cf2cb526914f04286b23c5489a09c0860e7c48a244495efce578b9048bf3143ac8e48de289b7cf5077e11d58d3e48c26f05df291ddddaabf4f4deb6b3294c3f18123bdcdf86d24ba04d2ffa338bac8acfa15c9b10707457d4037f8623ee7f14a80d53cd5a8195e6c0ef464f1111aef119f97adeeeec4f8e34b4d305b2f6975c55bfc839a2acf38c22a437abc51fc451716b61ce9aefd2a2ddf9f5060fe62fa349cc8c5ee201685f05f92a8581800000000000000a34559900a1e0269e434af2f7d675d01b74a066091a0267e20000000000000001a4c4b3dac184d3d7a6fd46bc165f3e1e43b96cdb8cef039787f6c6b7f8f7b7a'
  }
],
"status": "success"
}
```

if the client is initialized using using these keys.
```
ethereum pk:
Uint8Array(33) [
    2, 202, 118, 214,  19, 106,  39, 216,
  225, 169, 207,  51, 129, 179, 226,   0,
  109, 197,  49, 143, 238,   4, 214,  34,
  188, 182,  30,  67, 235,  13, 224, 104,
  147
]
ephemeral pk and sk:
Uint8Array(40) [
   32,   0,   0,   0,   0,   0,   0,   0,  39,
  121,  24, 184,  12, 127,  68, 154,  22, 227,
   23,   7,  21, 125, 147, 122,  93, 184, 218,
   95, 128,   0, 221,  28,  72, 202, 143, 130,
  120, 177, 161,  35
]
Uint8Array(40) [
   32,   0,   0,   0,   0,   0,   0,   0, 164,
  240, 151,  89, 111,  65, 111,  84, 125,  31,
  110, 146, 144, 255, 164,  60,  63, 124,  81,
  234, 131,  49,  39, 189, 190, 189, 223, 152,
  120, 188, 169, 103
]

```

the final result should be 48

Reply:
This is commented on purpose 😅 I didn't have any response from the gateway to use it as an example, so I just write fake test to check `fetch-mock` library, and I'll implement them later.